src/st_paywall/aggregate_auth.py

```diff
@@ -1,74 +1,52 @@
 import streamlit as st
-from .google_auth import get_logged_in_user_email, show_login_button
 from .stripe_auth import is_active_subscriber, redirect_button
 from .buymeacoffee_auth import get_bmac_payers
 
 payment_provider = st.secrets.get("payment_provider", "stripe")
 
 
-def add_auth(
-    required: bool = True,
-    login_button_text: str = "Login with Google",
-    login_button_color: str = "#FD504D",
-    login_sidebar: bool = True,
-
-):
+def add_auth(required: bool = True, show_redirect_button: bool = True, subscription_button_text: str = 'Subscribe now!', button_color: str = "#FD504D", use_sidebar: bool = True):
+    """Add authentication and payment verification to a Streamlit app."""
     if required:
-        require_auth(
-            login_button_text=login_button_text,
-            login_sidebar=login_sidebar,
-            login_button_color=login_button_color,
-        )
+        require_auth(show_redirect_button, subscription_button_text, button_color, use_sidebar)
     else:
-        optional_auth(
-            login_button_text=login_button_text,
-            login_sidebar=login_sidebar,
-            login_button_color=login_button_color,
-        )
-
+        optional_auth(show_redirect_button, subscription_button_text, button_color, use_sidebar)
 
-def require_auth(
-    login_button_text: str = "Login with Google",
-    login_button_color: str = "#FD504D",
-    login_sidebar: bool = True,
-):
-    user_email = get_logged_in_user_email()
 
-    if not user_email:
-        show_login_button(
-            text=login_button_text, color=login_button_color, sidebar=login_sidebar
-        )
+def require_auth(show_redirect_button: bool = True, subscription_button_text: str = 'Subscribe now!', button_color: str = "#FD504D", use_sidebar: bool = True):
+    """Require authentication and payment verification to proceed."""
+    if not st.experimental_user.is_logged_in:
         st.stop()
+
+    user_email = st.experimental_user.email
+
     if payment_provider == "stripe":
-        is_subscriber = user_email and is_active_subscriber(user_email)
+        is_subscriber = is_active_subscriber(user_email)
```

Comment:
It would be nice to go ahead and standardize these names a bit more, maybe `is_stripe_subscriber` and `is_bmac_subscriber`, or `is_subscriber(user_email, provider='stripe')`, etc.

Reply:
done :)